pkg/storage/stores/shipper/indexshipper/tsdb/head_manager.go

```diff
@@ -545,7 +546,7 @@ func legacyWalPath(parent string, t time.Time) string {
 
 // recoverHead recovers from all WALs belonging to some period
 // and inserts it into the active *tenantHeads
-func recoverHead(name, dir string, heads *tenantHeads, wals []WALIdentifier, legacy bool, logger log.Logger) error {
+func recoverHead(name, dir string, heads *tenantHeads, wals []WALIdentifier, legacy bool, logger log.Logger, metric *prometheus.CounterVec) error {
```

Comment:
I'd give the metric a more meaningfull name
```suggestion
func recoverHead(name, dir string, heads *tenantHeads, wals []WALIdentifier, legacy bool, logger log.Logger, repairsCounter *prometheus.CounterVec) error {
```

Reply:
Good point, handled.